### DISCUSSION TITLE
Move `config/address.c` to address library

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#3969] [Development]
The config code lives in the `config/` directory.
It is fairly self-contained; it only relies on helper functions from libmutt.

Except for two things: Config Cache, Address Type.

The architecture of the Config System should allow us to define types without needing the code to live in the `config/` directory.
Move the Address Type functions into the `address/` directory.

- `config/address.[ch]`
- Helper function from `config/helpers.[ch]`

This will remove libconfig's dependency on libaddress.

This is a simple job to adjust the Makefile, but there's slightly more to do, to connect everything up.

If you like puzzles, I'll leave it at that.
(or I can provide a list)

**Comment by roccoblues** [comment]
started here: https://github.com/neomutt/neomutt/pull/3992

**Comment by flatcap** [comment]
Completed in: f9d941ea1
